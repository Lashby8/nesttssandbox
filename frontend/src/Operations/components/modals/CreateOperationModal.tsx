import React, { PureComponent } from "react";
import { Modal, Box, Button, TextField } from "@mui/material";
import CloseIcon from "@mui/icons-material/Close";

export interface CreateOperationModalProps {
  isOpened: boolean;
  newOperationName: string;

  handleCloseAction: () => void;
  handleCreateNewOperation: () => void;
  handleChangeNewName: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export default class CreateOperationModal extends PureComponent<
  CreateOperationModalProps,
  Readonly<{}>
> {
  render() {
    const {
      isOpened,
      newOperationName,
      handleChangeNewName,
      handleCloseAction,
      handleCreateNewOperation,
    } = this.props;

    return (
      <Modal
        open={isOpened}
        onClose={handleCloseAction}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        <Box
          sx={{
            position: "absolute",
            top: "50%",
            left: "50%",
            transform: "translate(-50%, -50%)",
            width: 400,
            bgcolor: "background.paper",
            border: "2px solid #000",
            boxShadow: 24,
            p: 4,
            display: "flex",
            flexDirection: "column",
          }}
        >
          <Button
            sx={{ justifyContent: "flex-end" }}
            onClick={handleCloseAction}
          >
            <CloseIcon color="primary" />
          </Button>
          <TextField
            error={newOperationName.length === 0}
            helperText={"Should not be empty"}
            id="standard-basic"
            variant="standard"
            label="Insert a new name"
            value={newOperationName}
            onChange={handleChangeNewName}
          />
          <div className="modal-window_buttons" color="error">
            <Button onClick={handleCloseAction}>Cancel</Button>
            <Button onClick={handleCreateNewOperation}>Create</Button>
          </div>
        </Box>
      </Modal>
    );
  }
}
