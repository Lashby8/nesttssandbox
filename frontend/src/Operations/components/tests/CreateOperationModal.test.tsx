import React from "react";
import renderer from "react-test-renderer";
import CreateOperationModal, {
  CreateOperationModalProps,
} from "../modals/CreateOperationModal";

test("Closed CreateOperationModal", () => {
  const createOpModalProps: CreateOperationModalProps = {
    isOpened: false,
    handleCloseAction: jest.fn(),
    handleCreateNewOperation: jest.fn(),
    newOperationName: "",
    handleChangeNewName: jest.fn(),
  };
  const component = renderer.create(
    <CreateOperationModal {...createOpModalProps} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Opened CreateOperationModal", () => {
  const createOpModalProps: CreateOperationModalProps = {
    isOpened: false,
    handleCloseAction: jest.fn(),
    handleCreateNewOperation: jest.fn(),
    newOperationName: "This is totally new opeartion",
    handleChangeNewName: jest.fn(),
  };
  const component = renderer.create(
    <CreateOperationModal {...createOpModalProps} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
