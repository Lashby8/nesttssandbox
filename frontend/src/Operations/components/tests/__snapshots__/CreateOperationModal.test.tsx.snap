// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Closed CreateOperationModal 1`] = `null`;

exports[`Opened CreateOperationModal 1`] = `null`;
